drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DataManagementStrategy.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.editor.strategies;
+
+import org.drools.workbench.screens.scenariosimulation.client.models.ScenarioGridModel;
+import org.drools.workbench.screens.scenariosimulation.client.rightpanel.RightPanelView;
+import org.drools.workbench.screens.scenariosimulation.model.ScenarioSimulationModel;
+import org.drools.workbench.screens.scenariosimulation.model.ScenarioSimulationModelContent;
+import org.uberfire.backend.vfs.ObservablePath;
+
+/**
+ * The <b>Strategy</b> to use to manage/modify/save data inside the editor.
+ * Every actual implementation should provide methods to manage a specific kind of data source (ex. DRL, DMN)
+ */
+public interface DataManagementStrategy {
+
+
+    void populateRightPanel(final RightPanelView.Presenter rightPanelPresenter, final ScenarioGridModel scenarioGridModel);
+
+    ScenarioSimulationModel getModel();
+
+    void setModel(ScenarioSimulationModel model);
```

Comment:
Why does `DataManagementStrategy` contain `ScenarioSimulationModel`?
I would like to keep this component as much "stateless" as possible to avoid another reference to the model to be updated in case of undo/redo or similar action.
Wdyt?